benchmarking/runs/csv_weight_run_revised/weight_run_revised.py

```diff
@@ -0,0 +1,159 @@
+import csv
+import os
+from math import ceil
+from os import path
+from typing import List
+
+import typer
+
+from api.ai.interfaces.algorithm_config import PriorityAlgorithmConfig
+from api.models.enums import AlgorithmType, DiversifyType, ScenarioAttribute
+from benchmarking.evaluations.goals import DiversityGoal, WeightGoal
+from benchmarking.evaluations.interfaces import Goal, Scenario
+from benchmarking.evaluations.metrics.cosine_similarity import AverageCosineDifference
+from benchmarking.runs.csv_weight_run_revised.attributes import Attributes
+from benchmarking.runs.csv_weight_run_revised.data_provider_revised import DataProvider
+from benchmarking.runs.interfaces import Run
+from benchmarking.simulation.insight import Insight
+from benchmarking.simulation.simulation_set import SimulationSet, SimulationSetArtifact
+from benchmarking.simulation.simulation_settings import SimulationSettings
+
+
+class WeightRun(Run):
+    def start(self, num_trials: int = 1, generate_graphs: bool = True):
+        scenario = DefaultScenario()
+
+        # Metrics to be used for evaluation
+        metrics = [
+            AverageCosineDifference(
+                name="Score Cosine Difference",
+                attribute_filter=[Attributes.SCORE.value],
+            ),
+            AverageCosineDifference(
+                name="Timeslot Cosine Difference",
+                attribute_filter=[ScenarioAttribute.TIMESLOT_AVAILABILITY.value],
+            ),
+        ]
+        student_provider = DataProvider()
+        print(student_provider)
+        team_size = 4  # enter the desired team size
+        artifact: SimulationSetArtifact = SimulationSet(
+            settings=SimulationSettings(
+                num_teams=ceil(student_provider.num_students / team_size),
+                scenario=scenario,
+                student_provider=student_provider,
+                # cache_key=f"weight_run_for_bowen/weight_run_2/",
+                cache_key=f"csv_weight_run_revised/weight_run_revised/",
```

Comment:
it doesn't need a cache key at all, providing one saves it to the cache, leaving it undefined means the output is not saved.

In this instance, we don't need to cache (which means saving a .json of the output to your computer), because we're not planning to re-run it 100s of times for testing.